syncd/FlexCounter.cpp

```diff
@@ -3433,3 +3502,120 @@ void FlexCounter::notifyPoll()
     m_readyToPoll = true;
     m_pollCond.notify_all();
 }
+
+bool FlexCounter::isOnGearbox()
+{
+    SWSS_LOG_ENTER();
+
+    return m_dbCounters == "GB_COUNTERS_DB";
```

Comment:
this is not the best solution, since that db name could be changed to something else using config.json, this db name is default when no config is actually specified, maybe you could find a different way to figure out whether this is gearbox ? btw. why do you actually need to check this ?

Reply:
I have not found an easier and better solution. For default db name, that must not be a case for gbsyncd/gearbox, which always utilizes global context guid 1 in `/usr/share/sonic/hwsku/context_config.json`. Like `COUNTERS_DB` for asic, the name GB_COUNTERS_DB is generally common for gearbox.

Only on gbsyncd, it needs to run a specific plugin gb_port.lua for dealing with gearbox port counter. This plugin needs a list of tuples (portName, systemSidePortVid, lineSidePortVid) as KEYS.

Reply:
not sure what you meant by always utilizing guid 1, since that also don't seems to generic solution, using context_config.json, gearbox could be put anywhere, and it this will be some case in the future, this will cause some weird issues hard to find if some of those values will be hard coded :(